rts/Map/Generation/BlankMapGenerator.cpp

```diff
@@ -171,80 +201,40 @@ void CMapGenerator::GenerateMapInfo(CVirtualFile* fileMapInfo)
 	//Make info to put in mapinfo
 	std::stringstream ss;
 	std::string startPosString;
-	const std::vector<int2>& startPositions = GetStartPositions();
 	for (size_t x = 0; x < startPositions.size(); x++) {
 		ss << "[" << x << "] = {startPos = {x = " << startPositions[x].x << ", z = " << startPositions[x].y << "}},";
 	}
 	startPosString = ss.str();
 
 	//Replace tags in mapinfo.lua
 	luaInfo = StringReplace(luaInfo, "${NAME}", setup->mapName);
-	luaInfo = StringReplace(luaInfo, "${DESCRIPTION}", GetMapDescription());
+	luaInfo = StringReplace(luaInfo, "${DESCRIPTION}", mapDescription);
 	luaInfo = StringReplace(luaInfo, "${START_POSITIONS}", startPosString);
 
 	//Copy to filebuffer
 	fileMapInfo->buffer.assign(luaInfo.begin(), luaInfo.end());
 }
 
-void CMapGenerator::GenerateSMT(CVirtualFile* fileSMT)
+void CBlankMapGenerator::GenerateSMT(CVirtualFile* fileSMT)
 {
-	constexpr int32_t tileSize = 32;
-	constexpr int32_t tileBPP = 3;
-
 	//--- Make TileFileHeader ---
 	TileFileHeader smtHeader;
 	std::strcpy(smtHeader.magic, "spring tilefile");
 	smtHeader.version = 1;
-	smtHeader.numTiles = 1; //32 * 32 * (generator->GetMapSize().x * 32) * (generator->GetMapSize().y * 32);
-	smtHeader.tileSize = tileSize;
+	smtHeader.numTiles = 1; //32 * 32 * (generator->mapSize.x * 32) * (generator->mapSize.y * 32);
+	smtHeader.tileSize = 32;
```

Comment:
Where does this magic value come from? If the map has no texture can't it be a 512x512 tile or 1x1 tile, does it make a difference?

Reply:
> Where does this magic value come from? If the map has no texture can't it be a 512x512 tile or 1x1 tile, does it make a difference?

I think it doesn't. This part of code I just left as it was. I think it makes sense to make it 1x1.

Reply:
> Where does this magic value come from? If the map has no texture can't it be a 512x512 tile or 1x1 tile, does it make a difference?

But maybe this is not so good idea.
For some reason tileSize has a notation that it must be 32:
https://github.com/beyond-all-reason/spring/blob/0ee88788931f9f0b195eb5f895f1092fde4211c0/rts/Map/SMF/SMFFormat.h#L56

Reply:
Also this line:
https://github.com/beyond-all-reason/spring/blob/0ee88788931f9f0b195eb5f895f1092fde4211c0/rts/Map/SMF/SMFFormat.h#L118
Yea, I guess we should not touch that for now.